odoo/addons/test_orm/tests/test_search.py

```diff
@@ -0,0 +1,1202 @@
+            {'name': 'Belgium'}, {'name': 'France'},
+        ])
+        cls.brussels, cls.paris = cls.env['test_orm.city'].create([
+            {'name': "Brussels", 'country_id': cls.belgium.id},
+            {'name': "Paris", 'country_id': cls.france.id},
+        ])
+
+    def test_flush_fields_in_domain(self):
+        with self.assertQueries(['''
+            UPDATE "test_orm_city"
+            SET "name" = "__tmp"."name"::VARCHAR,
+                "write_date" = "__tmp"."write_date"::timestamp,
+                "write_uid" = "__tmp"."write_uid"::int4
+            FROM (VALUES %s) AS "__tmp"("id", "name", "write_date", "write_uid")
+            WHERE "test_orm_city"."id" = "__tmp"."id"
+        ''', '''
+            SELECT "test_orm_city"."id"
+            FROM "test_orm_city"
+            WHERE "test_orm_city"."name" LIKE %s
+            ORDER BY "test_orm_city"."id"
+        ''']):
+            self.brussels.name = "Bruxelles"
+            self.model.search([('name', 'like', 'foo')], order='id')
+
+    def test_flush_fields_in_subdomain(self):
+        with self.assertQueries(['''
+            UPDATE "test_orm_city"
+            SET "country_id" = "__tmp"."country_id"::int4,
+                "write_date" = "__tmp"."write_date"::timestamp,
+                "write_uid" = "__tmp"."write_uid"::int4
+            FROM (VALUES %s) AS "__tmp"("id", "country_id", "write_date", "write_uid")
+            WHERE "test_orm_city"."id" = "__tmp"."id"
+        ''', '''
+            SELECT "test_orm_city"."id"
+            FROM "test_orm_city"
+            WHERE "test_orm_city"."country_id" IN (
+                SELECT "test_orm_country"."id"
+                FROM "test_orm_country"
+                WHERE "test_orm_country"."name" LIKE %s
+            )
+            ORDER BY "test_orm_city"."id"
+        ''']):
+            self.brussels.country_id = self.france
+            self.model.search([('country_id.name', 'like', 'foo')], order='id')
+
+        with self.assertQueries(['''
+            UPDATE "test_orm_country"
+            SET "name" = "__tmp"."name"::VARCHAR,
+                "write_date" = "__tmp"."write_date"::timestamp,
+                "write_uid" = "__tmp"."write_uid"::int4
+            FROM (VALUES %s) AS "__tmp"("id", "name", "write_date", "write_uid")
+            WHERE "test_orm_country"."id" = "__tmp"."id"
+        ''', '''
+            SELECT "test_orm_city"."id"
+            FROM "test_orm_city"
+            WHERE "test_orm_city"."country_id" IN (
+                SELECT "test_orm_country"."id"
+                FROM "test_orm_country"
+                WHERE "test_orm_country"."name" LIKE %s
+            )
+            ORDER BY "test_orm_city"."id"
+        ''']):
+            self.belgium.name = "Belgique"
+            self.model.search([('country_id.name', 'like', 'foo')], order='id')
+
+    def test_flush_auto_join_field_in_domain(self):
+        self.patch(type(self.brussels).country_id, 'auto_join', True)
```

Comment:
use `self.env.registry['test_new_api.city']` to avoid the security check complaining it in the future